src/main/java/org/quickfixj/ilink3/ILink3Connector.java

```diff
@@ -76,279 +82,443 @@ public class ILink3Connector {
-		.firmId(settings.getString(SESSION_ID_ILINK3, SETTING_FIRM_ID))
-		.userKey(settings.getString(SESSION_ID_ILINK3, SETTING_USER_KEY))
-		.accessKeyId(settings.getString(SESSION_ID_ILINK3, SETTING_ACCESS_KEY_ID))
-		.tradingSystemName(settings.getString(SESSION_ID_ILINK3, SETTING_TRADING_SYSTEM_NAME))
-		.tradingSystemVendor(settings.getString(SESSION_ID_ILINK3, SETTING_TRADING_SYSTEM_VENDOR))
-		.tradingSystemVersion(settings.getString(SESSION_ID_ILINK3, SETTING_TRADING_SYSTEM_VERSION))
-		.handler(connectionHandler).requestedKeepAliveIntervalInMs(30000)
-//		.reEstablishLastConnection(true)
-		.build();
-
-	while (!library.isConnected()) {
-	    idleStrategy.idle(library.poll(1));
-	}
-
-	final Reply<ILink3Connection> reply = library.initiate(connectionConfiguration);
-
-	LOG.info("Executing connection initiation...");
-	while (reply.isExecuting()) {
-	    idleStrategy.idle(library.poll(1));
-	}
-
-	libraryPollingFuture = pollingExecutor.submit(new LibraryPollTask(library, idleStrategy));
-
-	if (reply.hasCompleted()) {
-	    connection = reply.resultIfPresent();
-	    LOG.info("Connected: " + connection);
-	} else if (reply.hasErrored()) {
-	    LOG.error("Error when connecting: " + reply.error());
-	} else if (reply.hasTimedOut()) {
-	    LOG.error("Timed out when connecting: " + reply);
-	}
+    public synchronized Message sendToTarget(final Message fixMessage) {
+
+        if (fixMessage == null) {
+            LOG.error("sendToTarget() called with NULL message");
+            return null;
+        }
+        Message sentMessage;
+
+        if (canSend()) {
+            synchronized (WRITE_LOCK) {
+                try {
+                    sentMessage = ILink3MessageConverter.convertFromFIXAndSend(fixMessage, connection);
+                } catch (Exception e) {
+                    LOG.error("Encountered exception when trying to send message {}", fixMessage, e);
+                    connection.abort();
+                    return null;
+                }
+                // TODO insert call to callback to enable modification of message before sending
+
+                if (messageStore != null) {
+                    try {
+
+                        LOG.info("Start Checking senderseqnums: store:{} connection:{}", messageStore.getNextSenderMsgSeqNum(), connection.nextSentSeqNo());
+                        messageStore.incrNextSenderMsgSeqNum();
+                        LOG.info("End Checking senderseqnums: store:{} connection:{}", messageStore.getNextSenderMsgSeqNum(), connection.nextSentSeqNo());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                } else {
+                    LOG.error("messageStore is null, cannot increment seqnum?");
+                }
+
+                fixMessageHandler.onFIXMessageSend(sentMessage);
+                connection.commit();
+                return sentMessage;
```

Comment:
Do we really need to return the message AND call `OnFIXMessageSent()`?
Normally (on vanilla QFJ) `sendToTarget` returns a `boolean`. Could we stick to the old semantics or is it required to also return a message here?